### PR TITLE
lecture-01 exercises / golang-tour pt2

### PR DESCRIPTION
**Pull request by marcello33**

### CODE REVIEW

File: lecture-01/slices/main.go [hunk 1]
```diff
@@ -107,3 +149,65 @@ func mergeUniqueInts(slices ...[]int) []int {
...
+
+// removeSameInts removes elements from 'dest' slice when they exist in 'source' slice
+func removeSameInts(dest, source []int) []int {
+	m := make(map[int]bool)
+
+	for _, val := range source {
+		m[val] = true
+	}
+
+	for i := 0; i < len(dest); i++ {
```
Comment (JekaMas) [inline lecture-01/slices/main.go]: Do you need that loop or it's possible just do something with the map?

File: lecture-01/slices/main.go [hunk 2]
```diff
@@ -107,3 +149,65 @@ func mergeUniqueInts(slices ...[]int) []int {
 
 	return res
 }
+
+// removeSameInts removes elements from 'dest' slice when they exist in 'source' slice
+func removeSameInts(dest, source []int) []int {
+	m := make(map[int]bool)
```
Comment (JekaMas) [inline lecture-01/slices/main.go]: map[int] struct {} ?

File: lecture-01/slices/main.go [hunk 3]
```diff
@@ -107,3 +149,65 @@ func mergeUniqueInts(slices ...[]int) []int {
...
+	for j := 1; j < len(s); j += 2 {
+		s[j-1], s[j] = s[j], s[j-1]
+	}
+
+	return s
+}
+
+// sortSlice sorts any slice of ints or strings in direct or reverse order
+func sortSlice[T int | string](s []T, reverse bool) []T {
+	sort.Slice(s, func(i, j int) bool {
```
Comment (JekaMas) [inline lecture-01/slices/main.go:206]: If you'd like to use standard libraty https://pkg.go.dev/sort#Reverse

Comment (marcello33) [reply lecture-01/slices/main.go:206]: Yeah, know that one, was trying to play around with some handmade stuff :D

File: lecture-01/slices/main.go (hunk 1 above)
Comment (marcello33) [reply lecture-01/slices/main.go]: hint? 👀

File: lecture-01/slices/main.go (hunk 2 above)
Comment (marcello33) [reply lecture-01/slices/main.go]: done

**Pull request merged**